pylint/checkers/imports.py

```diff
@@ -914,11 +914,27 @@ def _add_imported_module(self, node: ImportNode, importedmodname: str) -> None:
 
     def _check_preferred_module(self, node: ImportNode, mod_path: str) -> None:
         """Check if the module has a preferred replacement."""
-        if mod_path in self.preferred_modules:
+
+        mod_compare = [mod_path]
+        # build a comparison list of possible names using importfrom
+        if isinstance(node, astroid.nodes.node_classes.ImportFrom):
+            mod_compare = [f"{node.modname}.{name[0]}" for name in node.names]
+
+        # find whether there are matches with the import vs preferred_modules keys
+        matches = [
+            key
+            for key in list(self.preferred_modules.keys())
+            for compare_val in mod_compare
+            # determine if key is in compare_val
+            if key in compare_val
+        ]
```

Comment:
```suggestion
        matches = [
            k
            for k in self.preferred_modules
            for mod in mod_compare
            if k in mod
        ]
```
Shouldn't this work ? when iterating on a dict the default is ``keys()`

Reply:
Thank you, yes I believe you're correct, and this looks much cleaner. I'll add this into a forthcoming commit.